src/component/handlers/edit/editOnPaste.js

```diff
@@ -124,6 +126,13 @@ function editOnPaste(e: SyntheticClipboardEvent): void {
         );
         return;
       }
+    // Safari does not properly store text/html. Use the internalClipboard
+    // if present.   
+    } else if (isSafari && internalClipboard) {
```

Comment:
How do we know that the `internalClipboard` is what the user intended to paste? The code above checks to make sure that the pasted content is a match for our internal clipboard, but this doesn't do the same check.

So with this change, I might paste something from an external source after having earlier copy-pasted internally, and my new paste would be discarded in favor of the older internal clipboard.